### PR TITLE
[release] New directory for latest release performance metrics

### PR DESCRIPTION
**Pull request by khluu**
- Content copied over from `release/release_logs/2.22.0/`.
- Add `release_version.txt` file to show the latest release version that these perf are collected from.
- This directory will be keeping the latest release performance metrics, which would be updated for every releases instead of adding new version every time. `release/release_logs/` would be deprecated as a result

### CODE REVIEW

File: release/perf_metrics/release_version.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+2.22.0
```
Comment (aslonnie) [inline release/perf_metrics/release_version.txt:1]: maybe also use json (as all other files are json)? like a file called `version.json` or `meta.json` ?

which will also make it more expandable, where we can add other meta info into the struct.

Comment (khluu) [reply release/perf_metrics/release_version.txt]: ohh that sounds better. I updated

**Review by can-anyscale — APPROVED** [review]

**Comment by aslonnie** [comment]
@khluu could you follow up and delete the (per-version) files at the old location?

**Pull request merged**